### PR TITLE
 feature/sg-223-segmentation-dataset-refactor

### PR DESCRIPTION
**Pull request by danbochman**
Refactored SegmentationDataset and made CityscapesDatasetInterface backwards compatible

### CODE REVIEW

File: src/super_gradients/training/datasets/segmentation_datasets/segmentation_dataset.py [hunk 1]
```diff
@@ -77,26 +50,8 @@ def __init__(self, root: str, list_file: str = None, samples_sub_directory: str
...
-        self.image_mask_transforms_aug = image_mask_transforms_aug or default_image_mask_transforms_aug
-        # FIXME: CROP SIZE CANNOT BE PASSED WHEN LIST
-        if image_mask_transforms is None:
-            image_mask_transforms = transform.Compose([Rescale(short_size=self.crop_size),
-                                                       CropImageAndMask(crop_size=self.crop_size, mode="center")
-                                                       ])
-
-        self.image_mask_transforms = image_mask_transforms
+
+        self.transforms = transform.Compose(transforms) if transforms else []
```
Comment (shaydeci) [inline src/super_gradients/training/datasets/segmentation_datasets/segmentation_dataset.py:54]: move the if statement inside brackets.

**Review by shaydeci — CHANGES_REQUESTED** [review]
Minor comment.

File: src/super_gradients/training/datasets/segmentation_datasets/segmentation_dataset.py (hunk 1 above)
Comment (danbochman) [reply src/super_gradients/training/datasets/segmentation_datasets/segmentation_dataset.py]: 🤦 My bad. Must have lost this fix somewhere when I reverted/merged...

**Review by shaydeci — APPROVED** [review]
LGTM

**Pull request merged**